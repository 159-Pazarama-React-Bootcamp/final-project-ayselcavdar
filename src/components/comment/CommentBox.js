import styles from './CommentBox.module.css';
import PropTypes from 'prop-types';

const CommentBox = (props) => {
  const { comment, handleComment, submitCommentLine } = props;

  return (
    <>
      <div className={styles['comment-box']}>
        <textarea
          onChange={handleComment}
          value={comment}
          rows="2"
          cols="35"
          placeholder="başvuruyu cevapla"
        />
        <span
          className={comment.length > 3 ? styles['active'] : styles['disable']}
          onClick={submitCommentLine}
        >
          Gönder
        </span>
      </div>
    </>
  );
};

export default CommentBox;

CommentBox.propTypes = {
  comment: PropTypes.string.isRequired,
  handleComment: PropTypes.func.isRequired,
  submitCommentLine: PropTypes.func.isRequired,
};
