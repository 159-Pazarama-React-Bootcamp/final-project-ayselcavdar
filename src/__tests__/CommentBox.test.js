import render from 'react-test-renderer';
import CommentBox from '../components/comment/CommentBox';

describe('<CommentBox /> rendering', () => {
  it('should render a textarea to type the comment', () => {
    const wrapper = shallow(
      <CommentBox
        handleComment={jest.fn()}
        submitCommentLine={jest.fn()}
        comment="hello"
      />
    );
    expect(wrapper.find('textarea')).toHaveLength(1);
  });

  it('should render only one span to send answer to the applicant.', () => {
    const wrapper = shallow(
      <CommentBox
        handleComment={jest.fn()}
        submitCommentLine={jest.fn()}
        comment="hello"
      />
    );
    expect(wrapper.find('span')).toHaveLength(1);
  });

  it('CommentBox snapshots check', () => {
    const comp = render.create(
      <CommentBox
        handleComment={jest.fn()}
        submitCommentLine={jest.fn()}
        comment="hello"
      />
    );
    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
