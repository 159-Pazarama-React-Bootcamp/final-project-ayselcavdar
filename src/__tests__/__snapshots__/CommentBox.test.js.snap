// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentAdd /> rendering CommentBox render check 1`] = `
<div
  className="comment-box"
>
  <textarea
    cols="35"
    onChange={[MockFunction]}
    placeholder="başvuruyu cevapla"
    rows="2"
    value="hello"
  />
  <span
    className="active"
    onClick={[MockFunction]}
  >
    Gönder
  </span>
</div>
`;

exports[`<CommentBox /> rendering CommentBox snapshots check 1`] = `
<div
  className="comment-box"
>
  <textarea
    cols="35"
    onChange={[MockFunction]}
    placeholder="başvuruyu cevapla"
    rows="2"
    value="hello"
  />
  <span
    className="active"
    onClick={[MockFunction]}
  >
    Gönder
  </span>
</div>
`;

exports[`CommentBox render check 1`] = `
<div
  className="comment-box"
>
  <textarea
    cols="35"
    onChange={[MockFunction]}
    placeholder="başvuruyu cevapla"
    rows="2"
    value="comment"
  />
  <span
    className="active"
    onClick={[MockFunction]}
  >
    Gönder
  </span>
</div>
`;
